packages/addon/lua/gelly/mods/blood-mod/init.lua

```diff
@@ -2,26 +2,47 @@
 print("Blood mod loaded")
 
 local BLOOD_CONFIGS = {
-	[DMG_BULLET] = {
-		MinDensity = 200,
+    {
+        DamageFlags = bit.bor(DMG_BULLET, 8194),
+        MinDensity = 200,
 		MaxDensity = 400,
 		VelocityPower = 10, -- bullets usually are rotating so they can end up flinging blood
 		Randomness = 0.8, -- spray in the direction of the normal
 		CubeSize = 9,
 		DamageMultiplier = 15, -- density is added by the damage * this
-	},
-
-	[DMG_CRUSH] = {
-		MinDensity = 100,
+    },
+    {
+        DamageFlags = DMG_BLAST,
+        MinDensity = 3200,
+		MaxDensity = 5200,
+		VelocityPower = 40,
+		Randomness = 1,
+		CubeSize = 10,
+		DamageMultiplier = 90,
+    },
+	{
+        DamageFlags = bit.bor(DMG_SLASH, DMG_CLUB),
+        MinDensity = 100,
+		MaxDensity = 200,
+		VelocityPower = 4,
+		Randomness = 0.1,
+		CubeSize = 3,
+    },
+	{
+        DamageFlags = DMG_CRUSH,
+        MinDensity = 100,
 		MaxDensity = 200,
 		VelocityPower = 4,
 		Randomness = 1,
 		CubeSize = 15,
 		DamageMultiplier = 20,
-	},
+    },
+}
 
+local BLOOD_WEAPON_CONFIGS = { -- IMPLEMENT SPECIFIC WEAPON DAMAGE
 	-- double barrel shotgun in m9k
-	[4098] = {
+	{
+		ClassName = "m9k_dbarrel",
```

Comment:
For the damage flags we had to use an array instead of a lookup table due to needing to check the presence of the bit in the damage type flags rather than simple equality

This isn't the case for weapon classes which _is_ using equality below, so this can be made a lookup table for `O(1)` reads:
```lua
local WEAPON_BLOOD_CONFIGS = {
  m9k_dbarrel = { --[[ config here ]] }
}
```